### ISSUE TITLE
prefer_final_locals false positive in for-loop destructured records

### ISSUE THREAD

**Issue by Leptopoda** [dart-lang/linter#4832]
**Describe the issue**
Similar to: #4539

**To Reproduce**
Dart SDK version: 3.2.3 (stable) (Tue Dec 5 17:58:33 2023 +0000) on "linux_x64"
```dart
  for (var (a, b) in [(1, 2)]) {
    print(a);
    print(b);
  }
```

**Expected behavior**
lint does not fire

**Additional context**
Looks like the lint falsely fires when the parameter is not reassigned.
